### PR TITLE
[SPARK-19817][SQL] Make it clear that `timeZone` option is a general option in DataFrameReader/Writer.

### PR DESCRIPTION
**Pull request by ueshin**
## What changes were proposed in this pull request?

As timezone setting can also affect partition values, it works for all formats, we should make it clear.

## How was this patch tested?

Existing tests.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReaderWriterOptions.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+*/
+
+package org.apache.spark.sql
+
+import org.apache.spark.sql.catalyst.util.DateTimeUtils
+
+/**
+ * General option(s) for the DataFrameReader and DataFrameWriter.
+ */
+object DataFrameReaderWriterOptions {
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReaderWriterOptions.scala]: Let's remove this and always use `DateTimeUtils.TIMEZONE_OPTION`

**Comment by cloud-fan** [comment]
LGTM

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReaderWriterOptions.scala (hunk 1 above)
Comment (ueshin) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReaderWriterOptions.scala]: I see, I'll remove this.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala [hunk 2]
```diff
@@ -60,6 +60,8 @@ object DateTimeUtils {
   final val TimeZoneGMT = TimeZone.getTimeZone("GMT")
   final val MonthOf31Days = Set(1, 3, 5, 7, 8, 10, 12)
 
+  val TIMEZONE_OPTION = "timeZone"
```
Comment (gatorsmile) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala]: If we do it for `timeZone`, I think we should also move `dateFormat` and `timestampFormat`?

**Comment by gatorsmile** [comment]
Thanks! Merging to master.

**Pull request closed without merging**